sdk/eventgrid/Azure.Messaging.EventGrid/api/Azure.Messaging.EventGrid.netstandard2.0.cs

```diff
@@ -7,9 +7,7 @@ public static partial class BinaryDataExtensions
     }
     public partial class CloudEvent
     {
-        public CloudEvent(string source, string type) { }
-        public CloudEvent(string source, string type, System.BinaryData data, string dataContentType = null) { }
-        public CloudEvent(string source, string type, object data, string dataContentType = null) { }
+        public CloudEvent(string source, string type, object data = null, string dataContentType = null, System.Type dataSerializationType = null) { }
```

Comment:
From some early UX studies, we had an idea that many optional parameters scare customers. That's why we opted for overloads in commonly used cases. This seems to do the oposite.

Reply:
In general, users will be sending the data. By having a separate overload, users were not able to find the overload that contained the `data` parameter.

Reply:
``` C#
public CloudEvent(string source, string type, object data) 
public CloudEvent(string source, string type, object data = null, string dataContentType = null, System.Type dataSerializationType = null) 
```
??

Reply:
I'm not sure if this is better. Now we introduce the scenario where most users will not set dataContentType.

Reply:
I think having a single `public CloudEvent(string source, string type, object data = null, string dataContentType = null, System.Type dataSerializationType = null) { }
` overload doesn't serves the "In general, users will want to set the data" goal.

Reply:
Can we make data nullable but not optional?

Reply:
That is a possibility.

Reply:
Let's see what others think, but you are proposing:
```cs 
public CloudEvent(string source, string type, object data, string dataContentType = null, System.Type dataSerializationType = null) 
```
And we just allow null data, right?

Reply:
Or a pair of 
```
public CloudEvent(string source, string type, object data) 
public CloudEvent(string source, string type, object data, string dataContentType = null, System.Type dataSerializationType = null) 
```